tests/rados/pool_tests.py

```diff
@@ -1439,6 +1408,132 @@ def run(ceph_cluster, **kw):
+            if not rados_obj.create_erasure_pool(**profile_configs):
+                log.error("Failed to create the EC Pool")
+                return 1
+
+            rados_obj.bench_write(**profile_configs)
+            rados_obj.bench_read(**profile_configs)
+            log.info("Created the EC Pool and finished writing data into the pool.")
+
+            old_value = profile_configs["k"]
+            profile_configs["k"] += 1  # Increment the 'k' value
+            profile_configs["create_ecpool"] = False
+
+            # Modify profile without --force flag
+            log.info("Attempt 1: Modifying the profile without the --force flag")
+            try:
+                rados_obj.create_erasure_pool(**profile_configs)
+            except Exception as err:
+                log.info(
+                    f"Expected exception. Cannot modify the profile without --force flag.\n"
+                    f"Expected Error: {err}"
+                )
+
+            out = rados_obj.get_ec_profile_detail(
+                profile=profile_configs["profile_name"]
+            )
+
+            if out and int(out["k"]) != old_value:
+                log.error(
+                    f"Profile updated. Error! Current Value : {out['k']} , Expected : {old_value}"
+                )
+                test_pass = False
+
+            # Modify profile with --force flag
+            log.info("Attempt 2: Modifying the profile with the --force flag")
+            try:
+                rados_obj.create_erasure_pool(**profile_configs, force=True)
+            except Exception as err:
+                log.warning(f"Modification failed with --force flag. Exception: {err}")
+
+            # Check profile modification based on Ceph release
+            out = rados_obj.get_ec_profile_detail(
+                profile=profile_configs["profile_name"]
+            )
+            if out:
+                ceph_version = float(rhbuild.split("-")[0])
+                if ceph_version < 8.0 and int(out["k"]) != profile_configs["k"]:
+                    log.error(
+                        f"Profile updated with --force on build < 8.0. Build: {rhbuild}. Error!"
+                        f" Current Value : {out['k']} , Expected : {profile_configs['k']}"
+                    )
+                    test_pass = False
+                elif ceph_version >= 8.0:
+                    log.info(
+                        f"Retrying with --force and --yes-i-really-mean-it flags, build: {rhbuild}"
+                    )
+                    try:
+                        rados_obj.create_erasure_pool(
+                            **profile_configs, force=True, yes_i_mean_it=True
+                        )
+                    except Exception as err:
+                        log.info(
+                            f"Failed to modify profile with --force and --yes-i-really-mean-it flags. Error: {err}"
+                        )
```

Comment:
Should we not fail the test at this point?

Reply:
NO, we will not fail test here, and fail only if the profile was not actually not modified. 

Also, I was expecting the command failure to raise an exception , which is not happening. So we do not enter the except block. I have kept the except block just in case.